nixos/modules/services/desktops/espanso.nix

```diff
@@ -0,0 +1,40 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let cfg = config.services.espanso;
+in {
+  options = {
+    services.espanso = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable espanso
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.espanso;
+        description = ''
+          Espanso package to use
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.espanso = {
+      enable = true;
+      description = "Espanso daemon";
+      path = with pkgs; [ cfg.package libnotify xclip ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/espanso daemon";
+        Restart = "on-failure";
+      };
+      wantedBy = [ "default.target" ];
```

Comment:
Should we be adding something like `graphical.target` here?

Reply:
Adding it seems to be starting it too early in the process, which makes espanso error out with `espanso[5371]: Invalid M`.